lib/3lang/parser.md

```diff
@@ -0,0 +1,68 @@
+# 3bot parser
+
+- the objective of this parser is to be able to parse and execute actions from md files on different tf modules
+- we already have md file parser that could extract actions and their parameters.
+- these actions could be passed to the 3bot parser to parse and execute the user's requests
+- the 3bot parser should have smaller inner parsers for each tf module
+
+## Example
+
+- if a user wants to deploy 4 machines, it could go as follows:
+  
+```md
+    !!tfgrid.main.machines.deploy
+        name: 'project1'
+        network: private
+        ssh_key: 'ssh_key'
+        number_of_machines: 4
+        capacity: medium
+```
+
+- this would deploy 4 vms, with medium capacity (cru, mru, sru, hru) on the same network.
+- this is fairly simple, as we decide most of the specs for the user.
+- the problem comes when the user wants to have customizable workloads, as the md parser does not recognize nested structures. this requires some helper actions to assist in executing main actions.
```

Comment:
I think if the user wants customizable workloads he shouldn't be using the 3lang but rather use the V client lib or the send json rpc to the server directly.